packages/arb-token-bridge-ui/src/pages/api/withdrawals.ts

```diff
@@ -63,30 +59,40 @@ export default async function handler(
       })
     }
 
+    const additionalFilters = `${
+      typeof fromBlock !== 'undefined'
+        ? `l2BlockNum_gte: ${Number(fromBlock)},`
+        : ''
+    }
+    ${
+      typeof toBlock !== 'undefined'
+        ? `l2BlockNum_lte: ${Number(toBlock)},`
+        : ''
+    }
+    ${search ? `l2TxHash_contains: "${search}"` : ''}
+    `
+
     const subgraphResult = await getL2SubgraphClient(Number(l2ChainId)).query({
       query: gql`{
             withdrawals(
-                where: {
-                ${sender ? `sender: "${sender}",` : ''}
-                ${senderNot ? `sender_not: "${senderNot}",` : ''}
-                ${receiver ? `receiver: "${receiver}",` : ''}
-                ${receiverNot ? `receiver_not: "${receiverNot}",` : ''}
-                ${
-                  typeof fromBlock !== 'undefined'
-                    ? `l2BlockNum_gte: ${Number(fromBlock)}`
-                    : ''
-                }
+              where: {            
+                or: [
+                  ${
+                    sender
+                      ? `{ sender: "${sender}", ${additionalFilters} },`
+                      : ''
+                  }
                   ${
-                    typeof toBlock !== 'undefined'
-                      ? `l2BlockNum_lte: ${Number(toBlock)}`
+                    receiver
+                      ? `{ receiver: "${receiver}", ${additionalFilters} },`
                       : ''
-                  }  
-                  ${search ? `l2TxHash_contains: "${search}"` : ''}
-                }
-                orderBy: l2BlockTimestamp
-                orderDirection: desc
-                first: ${Number(pageSize)},
-                skip: ${Number(page) * Number(pageSize)}
+                  }
+                ]
+              }
+              orderBy: l2BlockTimestamp
```

Comment:
I'm not 100% sure if withdrawals have a timestamp prop, but if they do I will change it.

Although it won't make much difference to us because we sort them in the UI as well.

Reply:
Don't think we can, unless I'm wrong, we can only fetch timestamp from logs which we do, and subgraphs don't have this info